libminifi/include/core/ProcessContext.h

```diff
@@ -135,6 +135,10 @@ class ProcessContext : public controller::ControllerServiceLookup, public core::
     return getProperty(property.name, value);
   }
 
+  virtual bool getProperty(bool /*supports_expression_language*/, std::string_view property_name, std::string& value, const std::shared_ptr<FlowFile>& /*flow_file*/) {
+    return getProperty(property_name, value);
+  }
```

Comment:
Why do we need a new virtual overload? I'd prefer a single virtual function as an customization point, and possibly other non-static overloads to handle different use cases, to keep things simple.

Reply:
We need an overload  with the `flow_file` parameter that only needs the `property_name` as that is the only one available and this version was already available as a private function. It is probably better to create a new `Property` constructor overload with these parameters, wrap these and pass it to the already available `getProperty` overload.

Reply:
Updated in e8757f9c370b2ec92866a3731af1e6d9ae0edb74